src/plugins/intel_gpu/src/plugin/ops/pooling.cpp

```diff
@@ -97,7 +113,57 @@ static void CreateMaxPoolOp(Program& p, const std::shared_ptr<ngraph::op::v1::Ma
     p.AddPrimitiveToProfiler(op);
 }
 
+static void CreateMaxPoolOp(Program& p, const std::shared_ptr<ngraph::op::v8::MaxPool>& op) {
+    p.ValidateInputs(op, {1});
+    if (op->get_output_size() != 2) {
+        IE_THROW() << "MaxPool opset 8 requires 2 outputs";
+    }
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    std::string layerName = layer_type_name_ID(op) + ".0";
+
+    auto mutable_precision = op->get_output_element_type(1);
+    cldnn::layout mutableLayout = cldnn::layout(DataTypeFromPrecision(mutable_precision),
+                                                DefaultFormatForDims(op->get_output_shape(1).size()),
+                                                CldnnTensorFromIEDims(op->get_output_shape(1)));
+    auto shared_memory = p.GetEngine().allocate_memory(mutableLayout);
+    cldnn::primitive_id maxpool_mutable_id_w = layer_type_name_ID(op) + "_md_write";
+    auto indices_mutable_prim = cldnn::mutable_data(maxpool_mutable_id_w,
+                                                   shared_memory,
+                                                   op->get_friendly_name());
```

Comment:
`op->get_friendly_name()` potentially could be non-trivial and is called 3 times here